### PR TITLE
Revert "river: refactor"

### PR DESCRIPTION
**Pull request by heavyrain266**
Reverts NixOS/nixpkgs#121357

Why revert?
- Originally packages in header are sorted by phases for easier dependency management (by libs etc.)
- Run hooks in prebuild gives me segfault locally (tested on NixOS and kiss linux with nix)
- No needed for super long description about design goals copied straight from project's readmie
- Derivation was built twice because of `zig build` in preBuild phase
- Looks like previous commiter doesn't know anything about zig and it's build system

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 121405` at 8776e9a6 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>river</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/applications/window-managers/river/default.nix:26:3:
```
   |
26 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

**Comment by jonringer** [comment]
cc @AndersonTorres

File: pkgs/applications/window-managers/river/default.nix [hunk 1]
```diff
@@ -24,62 +12,33 @@ stdenv.mkDerivation rec {
...
   installPhase = ''
-    runHook preInstall
-    zig build -Dman-pages -Drelease-safe -Dxwayland --prefix $out install
-    runHook postInstall
+    zig build -Drelease-safe -Dxwayland -Dman-pages --prefix $out install
   '';
 
+  nativeBuildInputs = [ zig wayland scdoc pkg-config ];
+
+  installFlags = [ "DESTDIR=$(out)" ];
```
Comment (AndersonTorres) [inline pkgs/applications/window-managers/river/default.nix:32]: It should not be needed. DESTDIR has [the same semantics](https://github.com/ziglang/zig/issues/2929) of autotools' DESTDIR variable; therefore, it is not useful for Nix. 

DESTDIR is useful for classic package managers/build scripts like Slackbuilds or PKGBUILDs.

**Comment by AndersonTorres** [comment]
It will be long...

> Derivation was built twice because of zig build in preBuild phase

That was my fault.

I just followed the [AUR](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=river-git) script here. It repeats the command adding the 'install' argument in the second invocation. 

It looks like the same principle of calling `make && make install` from classic projects. Following the idea that `make` does not rebuild targets already built, it appeared that Zig would do the same, not rebuilding already built targets.

Also, it looked strange conflating the build and install phases in a single one. This way, the AUR script looks plausible. 

But I should have looked the build logs. Indeed it rebuilds the whole tree. My fault, my maximum fault.

On the other hand, there is another way to make the expression more nixy: use `phases = [ "unpackPhase" "installPhase" ];`.

> Run hooks in prebuild gives me segfault locally (tested on NixOS and kiss linux with nix)

Hook on a prebuild? `preBuild` is a hook. I suppose you meant to say using `runHook` is creating segfaults; this is unexpected at least. Maybe a bug on nixpkgs or something?

On the other hand,, adding runHook is mandatory - look at the @r-rmcgibbo suggestion above.

**Comment by AndersonTorres** [comment]
> No needed for super long description about design goals copied straight from project's readmie

Here I will differ.

- A `longDescription` that just repeats the short `description`, without adding substantial information to it, is not useful.

  If a long description is to be included, there is no problem in copying sentences directly from upstream README. Indeed, this is the regular practice among nixpkgs expressions, and it conveys a useful introduction about the package ("minimalist, stacked", "rivertile and riverctl" etc.).

  Also, this is not superlong at all - it ocuppies less than 15% of the file.

  Else, just remove `longDescription` completely. A mere copy-paste of the first line of upstream README that has no substantial difference compared with the short `description` is useless.

**Comment by AndersonTorres** [comment]
For now I will revert.

**Comment by branwright1** [comment]
I will just add runHook after next update, for now just DESTDIR creates folder with manpages and 3 binaries in river derivation folder, without it won't work and that one mentioned in zig's github is included in nightly build where we are using 0.7.1 which is stable and miss a lot of features

**Comment by AndersonTorres** [comment]
But this is from an older release from 2019, namely, 0.5.0.

**Comment by branwright1** [comment]
None of your changes make sense, same as what you said comparing derivations to completelly different PKGBUILDs....

**Comment by jonringer** [comment]
> On the other hand, there is another way to make the expression more nixy: use `phases = [ "unpackPhase" "installPhase" ];`.

specific phases are discouraged, as phases such as "fixupPhase" still help with nixification (patching shebangs, etc). or at the very least, you should be including `patchPhase` and `fixupPhase`.

**Comment by AndersonTorres** [comment]
> comparing derivations to completelly different PKGBUILDs....

"Completely different"? 

Nixpkgs is not an "esoteric" system like Windows or OS/2. It is a package manager that expects a typical GNU environment and executes Bash scripts to perform its tasks. Hey, you quoted KISS Linux!

Things that affect other package managers in general affect Nixpkgs in a similar manner: deprecation of unmaintained software, unstable upstream dependencies, patches &c. Indeed, you can look at the source code of many derivations and notice that they use patches from [Arch](https://github.com/NixOS/nixpkgs/blob/8284fc30c84ea47e63209d1a892aca1dfcd6bdf3/pkgs/applications/misc/qlandkartegt/default.nix#L15), [Debian](https://github.com/NixOS/nixpkgs/blob/8284fc30c84ea47e63209d1a892aca1dfcd6bdf3/pkgs/games/rrootage/default.nix#L21), and even [Gentoo](https://github.com/NixOS/nixpkgs/blob/8284fc30c84ea47e63209d1a892aca1dfcd6bdf3/pkgs/development/libraries/xbase/default.nix#L14).

You assertion of "completely different" is unwarranted at least.

**Pull request merged**